whatshap/phred_scores.py

```diff
@@ -0,0 +1,89 @@
+"""
+Takes as input the chromosome wise ref-read kmer pair counts from whatshap learn,
+calculates the probability for observing each ref-read kmer pair across the whole genome
+and outputs a table with corresponding phred-scores
+"""
+
+import argparse
+import math
+from pathlib import Path
+import csv
+
+
+def phred(inputfile, outputfile, epsilon_value, kmer_size):
+    probs = {}
+    probabilities = {}
+    seen_comb = {}
+    comb_count = {}
+    k = int(kmer_size)
+    total_comb = float(4**k)
+    epsilon = float(epsilon_value)
+    path = Path(inputfile)
+    glob_path = path.glob("*.txt")
+    writer = open(outputfile, "w")
+    # Iterate over each chromosome file
+    for file in glob_path:
+        counts_file = open(file, "r")
+        reader = csv.reader(counts_file, delimiter="\t")
+        for line in reader:
+            ref = line[1]
+            read = line[2]
+            count = int(line[3])
+            new_key = (ref, read)
+            if new_key in probs:
+                probs[new_key] += count
+            else:
+                probs[new_key] = count
```

Comment:
You can avoid the "check if key is in dict and then update, else set" pattern by using a [defaultdict](https://docs.python.org/3/library/collections.html#collections.defaultdict).

Initialize probs thus:
```
probs = defaultdict(int)
```
then update:
```
probs[new_key] += count
```

(Same for `seen_comb`)